nomad/csi_endpoint.go

```diff
@@ -792,26 +802,39 @@ func (v *CSIVolume) controllerUnpublishVolume(vol *structs.CSIVolume, claim *str
 	} else if plugin == nil {
 		return fmt.Errorf("no such plugin: %q", vol.PluginID)
 	}
+
 	if !plugin.HasControllerCapability(structs.CSIControllerSupportsAttachDetach) {
+		claim.State = structs.CSIVolumeClaimStateReadyToFree
 		return nil
```

Comment:
If we skip the call to `checkpointClaim` does that mean that this claim state change is lost if a leadership transition happens?

Reply:
This assignment is really just for helping out testing. If we return `nil` from `controllerUnpublishVolume` the next step in the caller is to set `claim.State = structs.CSIVolumeClaimStateReadyToFree` and checkpoint.